src/index.test.ts

```diff
@@ -1715,3 +1720,438 @@ describe('READY event emission', () => {
+                storageProvider,
+                experimental: {
+                    togglesStorageTTL: 60,
+                },
+            };
+
+            const client = new UnleashClient(config);
+            await client.start();
+            expect(saveSpy).toHaveBeenCalledWith(
+                lastUpdateStorageKey,
+                {
+                    contextHash: expect.any(String),
+                    timestamp: expect.any(Number)
+                }
+            );
+            expect(saveSpy.mock.lastCall?.at(1).timestamp).toBeGreaterThanOrEqual(startTime);
+        });
+
+        test('Should store last update flag when fetch is successful with 304 status', async () => {
+            const startTime = Date.now();
+            fetchMock.mockResponseOnce(JSON.stringify(data), { status: 304 });
+
+            const config: IConfig = {
+                url: 'http://localhost/test',
+                clientKey: '12',
+                appName: 'web',
+                storageProvider,
+                experimental: {
+                    togglesStorageTTL: 60,
+                },
+            };
+
+            const client = new UnleashClient(config);
+            await client.start();
+            expect(saveSpy).toHaveBeenCalledWith(
+                lastUpdateStorageKey,
+                {
+                    contextHash: expect.any(String),
+                    timestamp: expect.any(Number)
+                }
+            );
+            expect(saveSpy.mock.lastCall?.at(1).timestamp).toBeGreaterThanOrEqual(startTime);
+        });
+
+        test('Should not store last update flag when fetch is not successful', async () => {
+            fetchMock.mockResponseOnce('', { status: 500 });
+
+            const config: IConfig = {
+                url: 'http://localhost/test',
+                clientKey: '12',
+                appName: 'web',
+                storageProvider,
+                experimental: {
+                    togglesStorageTTL: 60,
+                },
+            };
+
+            const client = new UnleashClient(config);
+            await client.start();
+            expect(saveSpy).not.toHaveBeenCalledWith(
+                lastUpdateStorageKey,
+                expect.any(Number)
+            );
+        });
+    });
+
+    describe('Handling last update flag storage hash value', () => {
+        let storageProvider: IStorageProvider;
+        let saveSpy: jest.SpyInstance;
+
+        class Store implements IStorageProvider {
+            public async save() {
+                return Promise.resolve();
+            }
+
+            public async get() {
+                return Promise.resolve([]);
+            }
+        }
+
+        beforeEach(() => {
+            storageProvider = new Store();
+            saveSpy = jest.spyOn(storageProvider, 'save');
+            jest.clearAllMocks();
+        });
+
+        test('Should compute last update flag value with the context value', async () => {
```

Comment:
On devrait éviter de tester l'implem, ce qu'on devrait plutôt tester c'est que les flags sont bien refresh si le contexte change (et ne le sont pas s'il ne change pas)

Reply:
Je vois que c'est testé plus bas donc on devrait supprimer ce test

Reply:
ok je vais modifier le test par contre ce que l'on fait plus bas (si on parle de "Should perform an initial fetch when context has changed, even if flags are up to date") ... ce n'est pas tout a fait pareil

dans le cas present, l'objectif c'est de tester que sur un update de contexte que les flags ne sont pas les memes (this.updatecontext)

dans l'autre, on est dans la phase d'initialisation de l'objet ... avec un contexte qui est different de ce que l'on avait avant dans le storage

Reply:
ha non tu devais parler de ce test : "Should perform a fetch when context is updated, even if flags are up to date'